### PR TITLE
OpenXR bindings

### LINKED ISSUE #193: OpenXR bindings

### PR DESCRIPTION
**Pull request by Perksey**
# Summary of the PR
Adds OpenXR bindings by reusing the Vulkan generator.

# Related issues, Discord discussions, or proposals
#48 
Closes #193 

# Further Comments
This also changes the OpenGL representation of strings to `byte*` from `char*` by popular demand.

### CODE REVIEW

File: src/OpenXR/Silk.NET.OpenXR/XrExtensionLoader.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                {
+                    throw new EntryPointNotFoundException
+                    (
+                        $"Entry point \"{functionName}\" not found. Note that instance and device extensions must be " +
+                        "enabled in order to load symbols correctly."
+                    );
+                }
+            }
+
+            return ret;
```
Comment (HurricanKai) [inline src/OpenXR/Silk.NET.OpenXR/XrExtensionLoader.cs:55]: What is the point of this?
Simply return in the try, and throw in the catch (possibly setting the caught exception as internal exception?)

File: src/OpenXR/Silk.NET.OpenXR/XrLoader.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                    return sym;
+                }
+            }
+            catch
+            {
+                // do nothing, just move on.
+            }
+
+            OpenXR.GetInstanceProcAddr(default, symbolName, ref Unsafe.As<IntPtr, FuncPtr>(ref sym));
+            if (sym != IntPtr.Zero)
```
Comment (HurricanKai) [inline src/OpenXR/Silk.NET.OpenXR/XrLoader.cs:62]: nit: `default` > `IntPtr.Zero`

File: src/Assimp/Silk.NET.Assimp/AssimpLibraryNameContainer.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ internal class AssimpLibraryNameContainer : SearchPathContainer
         public override string IOS => "__Internal";
 
         /// <inheritdoc />
-        public override string Windows64 => "Assimp64.dll";
+        public override string Windows64 => "Assimp32.dll";
```
[the remark this answers is not in the document]
Comment (Perksey) [reply src/Assimp/Silk.NET.Assimp/AssimpLibraryNameContainer.cs:28]: Whoopsie!

```suggestion
        public override string Windows64 => "Assimp64.dll";
```

File: src/OpenXR/Silk.NET.OpenXR/XrLoader.cs (hunk 2 above)
Comment (Perksey) [reply src/OpenXR/Silk.NET.OpenXR/XrLoader.cs:62]: Will fix

```suggestion
            if (sym != default)
```

File: src/OpenXR/Silk.NET.OpenXR/XrLoader.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            if (sym != default)
+            {
+                return sym;
+            }
+
+            if (OpenXR.CurrentInstance.HasValue)
+            {
+                OpenXR.GetInstanceProcAddr
+                    (OpenXR.CurrentInstance.Value, symbolName, ref Unsafe.As<IntPtr, FuncPtr>(ref sym));
+                if (sym != IntPtr.Zero)
```
Comment (HurricanKai) [inline src/OpenXR/Silk.NET.OpenXR/XrLoader.cs:71]: nit: `default` > `IntPtr.Zero`
```suggestion
                if (sym != default)
```

**Comment by Perksey** [comment]
Thanks!

**Pull request merged**